Fixed qmake bugs.

4ctrl-alt-del opened the pull request:
While attempting to compile KINC on my local machine for unit testing validation I found several bugs in the qmake project files that caused building KINC to fail. I fixed these for you really quick and made this tiny pull request.

4ctrl-alt-del committed:
Fixed qmake bugs.

Fixed multiple bugs in the qmake project files which made compiling with
CUDA and especially installing the software almost impossible.

The isEmpty test was being incorrectly used for the CUDADIR variable.

The program.files variable for INSTALLS was hardwiring the build
directory for its source. So if someone compiled in another shadow
directory it would fail finding anything to install.

bentsherman commented on src/KINC.pri:
```diff
@@ -33,11 +33,11 @@ isEmpty(LINK_LAPACKE) { LINK_LAPACKE = 1 }
 }
 else {
-   CUDADIR = $$(CUDADIR)
+   CUDADIR = $$CUDADIR
```
What is the purpose of this change? This code is intended to read the `CUDADIR` environment variable or else use a default, it looks like the new code would just always use the default.

4ctrl-alt-del replied:
This was my mistake I did that wrong. Now it should work as I intended:

If the CUDADIR variable is set via qmake command line, it is left alone
Else it is set to the CUDADIR environment variable if one exists
Else it is set to a default

4ctrl-alt-del committed:
Fixed CUDA directory in project file.

Made a second fix to how the project file initializes the CUDA working
directory for linking. It can now be set from the qmake command or an
environment variable. Setting it from the command line overrides any set
environment variable. If neither is set a default is set.
